src/coreclr/jit/assertionprop.cpp

```diff
@@ -5428,6 +5429,25 @@ GenTree* Compiler::optAssertionProp_Update(GenTree* newTree, GenTree* tree, Stat
             if (parent != nullptr)
             {
                 parent->ReplaceOperand(useEdge, newTree);
+
+                // If the parent is a GT_IND and we replaced the child with a handle constant, we might need
+                // to mark the GT_IND as invariant. This is the same as what gtNewIndOfIconHandleNode() does.
+                // Review: should some kind of more general morphing take care of this?
+                // Should this share code with gtNewIndOfIconHandleNode()?
+
+                if (parent->OperIs(GT_IND) && newTree->IsIconHandle())
```

Comment:
I've seen this even without JitOptRepeat. it's possible we should always just invoke some `gtUpdateSideEffects` on parent here as there could be other cases like this, but it might be costly so

Reply:
I was hitting this in #88860 in the past, tried to update the flags here like this too but I was still hitting the issue from some other place.